### PR TITLE
To Do list: add & remove

### PR DESCRIPTION
**Pull request by ZewdieMc**
## Hey Reviewer :wave: ,
In this PR, I implemented the following [project requirements.](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md)
###  General requirements

-  There are [no linter errors](https://github.com/microverseinc/linters-config) :heavy_check_mark: 
-  Followed the [GitHub Flow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/github_flow.md) .
-  The documentation is [professional](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).

### HTML/CSS requirements

-  Followed [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md).
### JavaScript requirements

- Followed [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).

### Project requirements
- Removed all hardcoded items from the tasks array.
- Created a new JavaScript file for the new functionality.
- Implemented a function for adding a new task (add a new element to the array).
- Implemented a function for deleting a task (remove an element from the array).
- Implemented a function for editing task descriptions.
- By default new tasks have the property `completed` set to `false` and the property `index` set to the value of the new `array length `
- Deleting a task updates all remaining items' `indexes`, so they represent the current list order and are unique
- All changes to the To Do List are saved in local storage

### CODE REVIEW

**Review by EricMbouwe — CHANGES_REQUESTED** [review]
Hi @ZewdieMc, 

Good job so far 🔥!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 🔥🎉
- A function for adding a new task (add a new element to the array) is implemented. ✔
- A function for editing task descriptions is implemented. ✔
- A function for deleting a task (removing an element from the array) is implemented. ✔
- Style ✔
- No linter errors ✔

# Required Changes ♻️
- [ ] There is an issue with the display function. The app doesn't show any task after a refresh while there are tasks saved in the localStorage. Let's check it

Screenshot 2023-02-23 at 12 52 35 PM

- [ ] The clear All completed doesn't work as expected. Let's fix that

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ZewdieMc** [comment]
Hi @EricMbouwe  :wave: 
Thank you for your feedback.
I have [fixed the first required change](https://github.com/ZewdieMc/To-Do-List/pull/2/commits/52b7d2fe919c893852f1d345fc4145ffb4d39a0d).  However, the second required change is not for this project. It's for the [next project](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m2_interactive_list.md).

File: src/modules/displayList.js [hunk 1]
```diff
@@ -4,12 +4,13 @@ export default (todoObjects) => todoObjects.map((todo) => {
   li.setAttribute('data-index', todo.index);
   li.setAttribute('data-completed', todo.completed);
```
Comment (tugsem) [inline src/modules/displayList.js:5]: - [x] I couldn't check requested changes as your function for adding a new task isn't working right now. Kindly fix it. It gives this console error:
image

**Review by tugsem — CHANGES_REQUESTED** [review]
## STATUS: REQUIRED CHANGES ♻️
Hi @ZewdieMc, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are some issues that you still need to work on to go to the next project but you are almost there!

Alt Text

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by tugsem** [comment]
> Hi @EricMbouwe 👋 Thank you for your feedback. I have [fixed the first required change](https://github.com/ZewdieMc/To-Do-List/pull/2/commits/52b7d2fe919c893852f1d345fc4145ffb4d39a0d). However, the second required change is not for this project. It's for the [next project](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m2_interactive_list.md).

You are not responsible for that feature of the project yet. Kindly ignore it.

File: src/modules/displayList.js (hunk 1 above)
Comment (ZewdieMc) [reply src/modules/displayList.js:5]: Hi @tugsem :wave: 
Thank you for your feedback. 
Fixed, please [have a look](https://github.com/ZewdieMc/To-Do-List/pull/2/commits/0d46c44b4806367035bfa5b34974879a988d01fe).

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @ZewdieMc 

Your project is complete! Great job for making the changes requested by the previous reviewer🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request
- CRUD methods ok
- No linter errors

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

- [optional] please remove the live demo section since you don't yet have a deployed version of your application. this will make your readme more professional
- [optional] please your Commit messages should be meaningful.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**